hudi-common/src/main/java/org/apache/hudi/metadata/HoodieTableMetadataUtil.java

```diff
@@ -1458,8 +1462,12 @@ private static Comparable<?> coerceToComparable(Schema schema, Object val) {
   }
 
   private static boolean canCompare(Schema schema, HoodieRecordType recordType) {
+    return isColumnTypeSupported(schema, Option.of(recordType));
+  }
+
+  private static boolean isColumnTypeSupported(Schema schema, Option<HoodieRecordType> recordType) {
```

Comment:
also, if the record type is not set, we should got for the super set. 
i.e. 
the unsupported types are:
if its AVRO -> MAP
if record type is not set or if its SPARK -> MAP, RECORD, ARRAY